tests/fixtures/apply_rules/output_report/report_apply_property.json

```diff
@@ -62,8 +62,8 @@
       "message": "object_type=route, object_id=RERAB:2, property_name=route_geometry: property_old_value does not match the value found in the data"
     },
     {
-      "category": "ObjectNotFound",
-      "message": "object_type=route, object_id=RERAF: geometry unknown not found"
+      "category": "OldPropertyValueDoesNotMatch",
+      "message": "object_type=route, object_id=RERAF, property_name=route_geometry: property_old_value does not match the value found in the data"
```

Comment:
there is now dead code. https://github.com/CanalTP/transit_model/blob/master/src/apply_rules/property_rule.rs#L397

you may remove the code and add a comment that explains why this case should never happen now.

Reply:
I don't think it is dead code. The `geo_id` is given by an end-user when configuring an `apply-rules` process. So it is still possible to end up in this code with an incorrect ID I believe.

Reply:
This case in the link above means
an object has a geometry_id and there is no geometries in the collection of geometries with id = geometry_id
but this case never happen now with your change.

If I'm wrong you should write the test for this case

Reply:
I pushed a [unit test](https://github.com/CanalTP/transit_model/pull/543/files#diff-e3d3b1c0a718fe3ab0336f3ef26df712R1080-R1111) for testing this case.

In term of public API, we indeed accept a `Model` so the case you're mentioning should never happen. But there is also internal API, like `crate::apply_rules::property_rules::update_geometry()`, or `crate::apply_rules::property_rules::apply_rules()` (which accept a `Collections`, not a `Model`) that a future developer might use at some point.